pkg/gds/certman/certs_test.go

```diff
@@ -205,6 +205,166 @@ func (s *certTestSuite) TestCertManager() {
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceReminderRE,
+			Reason:    "reissuance_reminder",
+			Timestamp: callTime,
+		},
+		{
+			To:        s.conf.Email.AdminEmail,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ExpiresAdminNotificationRE,
+			Reason:    "expires_admin_notification",
+			Timestamp: callTime,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
+}
+
+func (s *certTestSuite) TestCertManagerSevenDayReissuanceReminder() {
+	_ = s.setupCertManager(sectigo.ProfileCipherTraceEE)
+	defer s.teardownCertManager()
+	defer s.fixtures.LoadReferenceFixtures()
+	require := s.Require()
+
+	// setup the datastore to contain the modified echoVASP
+	s.resetVASPDatabase()
+	echoVASP, err := s.fixtures.GetVASP("echo")
+	require.NoError(err, "could not get echo VASP")
+	echoVASP = s.setupVASP(echoVASP)
+
+	// Call the certman function at 6 days, which will send
+	// the seven day cert reissuance reminder to echoVASP.
+	s.updateVaspIdentityCert(echoVASP, 6)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(echoVASP.Id)
+	require.NoError(err)
+
+	// Ensure that the expected email has been sent, using
+	// the mock email client.
+	messages := []*emails.EmailMeta{
+		{
+			To:        v.Contacts.Technical.Email,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceReminderRE,
+			Reason:    "reissuance_reminder",
+			Timestamp: callTime,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
+}
+
+func (s *certTestSuite) TestCertManagerReissuance() {
+	_ = s.setupCertManager(sectigo.ProfileCipherTraceEE)
+	defer s.teardownCertManager()
+	defer s.fixtures.LoadReferenceFixtures()
+
+	// setup the datastore to contain the modified echoVASP
+	require := s.Require()
+	s.resetVASPDatabase()
+	echoVASP, err := s.fixtures.GetVASP("echo")
+	require.NoError(err, "could not get echo VASP")
+	echoVASP = s.setupVASP(echoVASP)
+
+	// Call the certman function at 8 days, which should
+	// reissue the VASP's identity certificate, send the
+	// email with the created pkcs12 password and send
+	// the whisper link, as well as notifying the TRISA
+	// admin that reissuance has started.
+	s.updateVaspIdentityCert(echoVASP, 8)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(echoVASP.Id)
+	require.NoError(err)
+
+	// Ensure that the expected email has been sent, using
+	// the mock email client.
+	messages := []*emails.EmailMeta{
+		{
+			Contact:   v.Contacts.Technical,
+			To:        v.Contacts.Technical.Email,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceStartedRE,
+			Reason:    "reissuance_started",
+			Timestamp: callTime,
+		},
+		{
+			To:        s.conf.Email.AdminEmail,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceAdminNotificationRE,
+			Reason:    "reissuance_admin_notification",
+			Timestamp: callTime,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
```

Comment:
Can we add a TODO here to test the email send logic? (e.g., if technical contact is not verified, email should be sent to the admin contact etc.)